src/realm/object-store/sync/app.cpp

```diff
@@ -157,6 +157,24 @@ HttpHeaders get_request_headers(const std::shared_ptr<User>& user, RequestTokenT
     return headers;
 }
 
+std::string trim_base_url(std::string base_url)
+{
+    while (base_url.back() == '/') {
```

Comment:
I know the base_url should not be empty, but it's undefined behaviour to call `back()` if `empty()` is true. we should add a guard.

Reply:
done.